Common/include/geometry/dual_grid/CPoint.hpp

```diff
@@ -857,4 +877,19 @@ class CPoint {
    */
   void SetIndex(unsigned long iPoint, bool input);
 
+  /*!
+   * \brief Set wall roughnesses according to stored closest wall information.
+   * \param[in] roughness - Mapping [rank][zone][marker] -> roughness
+   */
+  template<typename Roughness_type>
+  void SetWallRoughness(Roughness_type const& roughness){
+    for (unsigned long iPoint=0; iPoint<GlobalIndex.size(); ++iPoint) {
+      auto rankID = ClosestWall_Rank[iPoint];
+      auto zoneID = ClosestWall_Zone[iPoint];
+      auto markerID = ClosestWall_Marker[iPoint];
+      if(rankID >= 0){
```

Comment:
It's mainly safety. The `ClosestWall_Rank` is initialized by -1. If there is no viscous wall, it will stay -1. Right now this code is only called from `CGeometry::ComputeWallDistance`, and only if there is a viscous wall, so there is no immediate danger IMO. But I think we should keep it safe.